src/oauth2.rs

```diff
@@ -58,24 +59,17 @@ mod space_separated_scope {
             scope
                 .split_whitespace()
                 .map(|x| x.to_owned())
-                .collect::<Vec<String>>(),
```

Comment:
IMHO, it's unnessary to collect iterator into a Vec explicitly, the `from_iter` can construct from an iterator, but `cargo clippy` failed for missing the `collect` statement:

```log
error: usage of `FromIterator::from_iter`
Error:   --> src/oauth2.rs:58:12
   |
58 |           Ok(HashSet::from_iter(
   |  ____________^
59 | |             scope.split_whitespace().map(|x| x.to_owned()),
60 | |         ))
   | |_________^ help: use `.collect()` instead of `::from_iter()`: `scope.split_whitespace().map(|x| x.to_owned()).collect()`
   |
   = note: `-D clippy::from-iter-instead-of-collect` implied by `-D warnings`
   = help: for further information visit rust-lang.github.io/rust-clippy/master/index.html#from_iter_instead_of_collect

```

https://github.com/ramsayleung/rspotify/pull/185/checks?check_run_id=1899884101

Reply:
Ahhh I get it now. Yes, very smart. Give me a sec, I think I understand it.

Reply:
Done! Take a look and let me know.

Reply:
Great job :)

Reply:
Me too! :)